provider/datagen/src/transform/cldr/locale_canonicalizer/aliases.rs

```diff
@@ -145,7 +145,9 @@ impl From<&cldr_serde::aliases::Resource> for AliasesV1<'_> {
                         (language, None, Some(region), false)
                             if language == language!("sgn")
                                 && !replacement.language.is_empty()
-                                && replacement == replacement.language.as_str() =>
+                                && replacement.script.is_none()
+                                && replacement.region.is_none()
+                                && replacement.variants == subtags::Variants::new() =>
```

Comment:
Suggestion: `.variants` refers to `&[Variant]`, so you can call `.is_empty()` on it

```suggestion
                                && replacement.variants.is_empty() =>
```